paasta_tools/kubernetes_tools.py

```diff
@@ -784,10 +756,14 @@ def get_kubernetes_services_running_here_for_nerve(
     nerve_list = []
     for kubernetes_service in kubernetes_services:
         try:
-            registrations = read_all_registrations_for_service_instance(
-                kubernetes_service.name, kubernetes_service.instance, cluster, soa_dir,
+            kubernetes_service_config = load_kubernetes_service_config(
```

Comment:
If `load_kubernetes_service_config` is implemented like `load_marathon_service_config`, then this loop will re-parse kubernetes.yaml for every instance, which can cause significant slowdowns for large yaml files. Can you switch to PaastaServiceConfigLoader instead?

Reply:
I guess it will be a bit of repeated work for multiple instances of the same service. At least it's just those running on the particular host so I don't think it's that bad. I'm probably going to leave it be for now because I'm already changing a lot here and don't want to make it any riskier.